handler-simple/src/main/java/com/github/philippheuer/events4j/simple/ClassUtil.java

```diff
@@ -0,0 +1,33 @@
+package com.github.philippheuer.events4j.simple;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+class ClassUtil {
+
+    private static final BiConsumer<Class<?>, Set<Class<?>>> ADD_INTERFACES = (c, set) -> {
+        final Class<?>[] interfaces = c.getInterfaces();
+        for (Class<?> anInterface : interfaces) {
+            if (set.add(anInterface)) {
+                ClassUtil.ADD_INTERFACES.accept(anInterface, set);
+            }
+        }
+    };
+
+    static Collection<Class<?>> getInheritanceTree(final Class<?> clazz) {
```

Comment:
another possible optimization is to cache recent eventClass => inheritanceTree lookups, but probably not worth it for the simple module